sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/ResourceSpecification.java

```diff
@@ -10,7 +12,7 @@
  * A ResourceSpecification encapsulates a Mesos Resource that may be used by a Task and therefore specified in a
  * TaskSpecification.
  */
-@JsonDeserialize(as = DefaultResourceSpecification.class)
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME)
```

Comment:
This may be an incompatible change if this data is stored in zk. Not sure though.